pandas/tests/reshape/merge/test_merge.py

```diff
@@ -1370,6 +1370,22 @@ def f():
             household.join(log_return, how='outer')
         pytest.raises(NotImplementedError, f)
 
+    def test_merge_datetime_index(self):
```

Comment:
can you parameterize on ``ndarray, Series, Index`` (for the df.index.year arg)

Reply:
Done.